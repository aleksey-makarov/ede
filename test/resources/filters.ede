{
  "single": "Lorem ipsum-Phrase",
  "whitespace": "  A Word  ",
  "multi": "Navzdory všeobecnému\npřesvědčení\nLorem Ipsum není\nnáhodný text.",
  "bool": true,
  "number": 1.0,
  "alpha": ["a", "b", "c", "d", "e"],
  "numeric": [1, 2, 3, 4, 5],
  "object": {"foo": "x", "bar": "y"}
}
---
Boolean Operands:
{{ !bool }}
{{ !false }}
{{ bool && false }}
{{ true && bool }}
{{ bool || false }}
{{ false || bool }}

Eq:
Ord:
Num:
RealFrac:

Text Casing:
{{ single | lowerHead }}
{{ single | upperHead }}
{{ single | toLower }}
{{ single | toUpper }}
{{ single | toTitle }}
{{ single | toCamel }}
{{ single | toPascal }}
{{ single | toSnake }}
{{ single | toSpinal }}
{{ single | toTrain }}
{{ number | toOrdinal }}
{{ single | toLower | upperHead }}
{% if (single | toLower) == "lorem ipsum" %}
single | lower == "lorem ipsum"
{% endif %}

Text Substitution:
{{ single | takeWord }}
{{ single | dropWord }}
{{ single | dropUpper }}
{{ single | toCamel| dropLower }}
{{ single | stripPrefix("Lorem ") }}
{{ single | stripSuffix("Phrase") }}
{{ single | stripStart }}
{{ single | stripEnd }}
{{ single | replace("ipsum-", "Hi! ") }}
{{ single | remove("Phrase") }}

Text Indentation:
{{ multi | indentLines(1, false) }}
{{ multi | prependLines("...") }}
{{ single | justifyLeft(2, "<-") }}
{{ single | justifyRight(2, "<-") }}
{{ single | center(6, "-") }}

Sequence Length:
{{ "" | length }}
{{ "123" | length }}
{{ alpha | length }}
{{ numeric | length }}
{{ object | length }}
{{ "" | empty }}
{{ "supercalifragilistic" | empty }}
{{ alpha | empty }}
{{ numeric | empty }}
{{ object | empty }}

List Subdivision:
{{ multi | head }}
{{ alpha | head }}
{{ numeric | init | show }}
{{ alpha | tail | show }}
{{ alpha | last }}
{{ single | splitWords | last }}

List Sorting:
{{ single | reverse }}
{# {{ single | sort | show }} #}
{# {{ alpha | reverse | sort | show }} #}
{# {{ object | sort | show }} #}

List Transformations:
{# {{ alpha | zip(numeric) }} #}
{# [{{ alpha | join(", ") }}] #}
{# ({{ numeric | map(toOrdinal) | join(", ") }}) #}
{# {{ numeric | map(toOrdinal) }} #}
{# {{ numeric | filter(>= 3) }} #}

Polymorphic Show:
{{ single | show }}
{{ bool | show }}
{{ number | show }}
{{ alpha | show }}
{{ numeric | show }}
{{ object | show }}

Polymorphic Existence:
{# {{ single | defined }} #}
{# {{ nonexistent | defined }} #}
{# {{ nonexistent | default(true) }} #}
