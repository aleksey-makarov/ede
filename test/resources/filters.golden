Boolean Operands:
false
true
false
true
true
true

Eq:
Ord:
Num:
RealFrac:

Text Casing:
Text Casing:
lorem ipsum-Phrase
Lorem ipsum-Phrase
lorem ipsum-phrase
LOREM IPSUM-PHRASE
Lorem Ipsum Phrase
loremIpsumPhrase
LoremIpsumPhrase
lorem_ipsum_phrase
lorem-ipsum-phrase
Lorem-Ipsum-Phrase
1st
Lorem ipsum-phrase
single | lower == "lorem ipsum"

Text Substitution:
Lorem
ipsum-Phrase
orem ipsum-Phrase
IpsumPhrase
ipsum-Phrase
Lorem ipsum-
Lorem ipsum-Phrase
Lorem ipsum-Phrase
Hi!
Phrase

Sequences:
0
3
5
1
true
false
false
false

Collections:

Poly:
"Lorem Ipsum"
true
1.0
["a","b","c","d","e"]
{"key":"value"}
